R/teal_lockfile.R

```diff
@@ -105,10 +96,23 @@ teal_lockfile_downloadhandler <- function() {
   )
 }
 
-is_r_cmd_check <- function() {
-  ("CheckExEnv" %in% search()) || any(c("_R_CHECK_TIMINGS_", "_R_CHECK_LICENSE_") %in% names(Sys.getenv()))
+lockfile_status <- function(process) {
+  renv_logs <- process$read_output()
+  if (any(grepl("Lockfile written", renv_logs))) {
```

Comment:
1. This `if else` is incomplete. I made `renv` to fail and I printed the message:

<img width="646" alt="image" src="https://github.com/insightsengineering/teal/assets/6959016/928a0b61-6013-4f4f-8a5d-1da9b6c6a53d">
I think that success or failure should be followed by `showNotification("Lockfile created successfully.")` or `showNotification("Lockfile failed to create.")`


2. I think download lockfile shouldn't be shown when lockfile is not created.

Reply:
> I think download lockfile shouldn't be shown when lockfile is not created.

This simplified the code a bit, as we no longer need to wait once the download button is clicked but the process is not finished 6010c4fb1d08b9de8c4c52afafb398e66421570f